File: src/main/java/br/com/jiratorio/config/AuditorAwareImpl.java
package br.com.jiratorio.config;

import br.com.jiratorio.domain.vo.Account;
import java.util.Optional;
import org.springframework.data.domain.AuditorAware;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Component;

@Component
public class AuditorAwareImpl implements AuditorAware<String> {

    @Override
    public Optional<String> getCurrentAuditor() {
        return Optional
                .ofNullable(SecurityContextHolder.getContext().getAuthentication())
                .map(it -> (Account) it.getPrincipal())
                .map(Account::getUsername);
    }

}

File: src/main/java/br/com/jiratorio/config/AuditorAwareImpl.kt
package br.com.jiratorio.config

import br.com.jiratorio.domain.Account
import java.util.Optional
import org.springframework.data.domain.AuditorAware
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component

@Component
class AuditorAwareImpl : AuditorAware<String> {

    override fun getCurrentAuditor() =
        Optional.ofNullable(SecurityContextHolder.getContext().authentication)
            .map { it.principal as Account }
            .map { it.username }!!

}
